modules/core/src/main/java/org/apache/synapse/config/xml/LogMediatorFactory.java

```diff
@@ -84,9 +85,12 @@ public Mediator createSpecificMediator(OMElement elem, Properties properties) {
             }
         }
 
+        OMAttribute logFullPayload = elem.getAttribute(ATT_LOG_FULL_PAYLOAD);
         // Set the high level set of properties to be logged (i.e. log level)
         OMAttribute level = elem.getAttribute(ATT_LEVEL);
-        if (level != null) {
+        if (logFullPayload != null) {
```

Comment:
we are only checking the attribute is there, we need to check for "true" condition also

Reply:
Fixed